### PR TITLE
Feature qrcode

### LINKED ISSUE #130: QR Code
As a normal player, I want to be able to connect to the game by scanning a QR code, so I don't have to type in the entire IP address of the server.

This task has subtasks:
- #131 QR Code - Generate
- #132 QR Code - Get URL
- #133 QR Code - Show

### LINKED ISSUE #131: QR Code - Generate
Generate QR codes from a given URL.

This is a subtask of #130.

### LINKED ISSUE #132: QR Code - URL
Get the URL to connect to.

This is a subtask of #130 QR Code.

### PR DESCRIPTION
**Pull request by Diocruel**
QR codes can now be generated! Please refer to the use of the QRGenerator to the javadoc.

Cobertura line coverage is 85%, though it can be argued that these test do not contribute much to actual testing. In order for the QRGenerator to work there is an output file needed and the use of a static final class: the NetworkInterface class. This hard to mock so I think it is okay to do it this way. 

This item was done early so that @WytzeElhorst can work on his sprint items.

This closes #130  , closes #131  , closes #132

### CODE REVIEW

File: src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+					InetAddress addr = a.nextElement();
+					String hostAddress = addr.getHostAddress();
+					if ((!hostAddress.startsWith("127.") 
+							&& !hostAddress.contains(":"))
+							&& !hostAddress.startsWith("192.168.")) {
+						hostingAddress = addr.getHostAddress();
+					}
+				}
+			}
+			hostingAddress = "https://" + hostingAddress + ":" + portNumber + "/";
```
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java]: I don't think we even support HTTPS, but not sure.

File: src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+			while (networkInterfaces.hasMoreElements()) {
+				NetworkInterface e = networkInterfaces.nextElement();
+				Enumeration<InetAddress> a = e.getInetAddresses();
+
+				while (a.hasMoreElements()) {
+					InetAddress addr = a.nextElement();
+					String hostAddress = addr.getHostAddress();
+					if ((!hostAddress.startsWith("127.") 
+							&& !hostAddress.contains(":"))
+							&& !hostAddress.startsWith("192.168.")) {
```
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java]: I don't think we want to exclude 192.168.

File: src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+			networkInterfaces = NetworkInterface.getNetworkInterfaces();
+			while (networkInterfaces.hasMoreElements()) {
+				NetworkInterface e = networkInterfaces.nextElement();
+				Enumeration<InetAddress> a = e.getInetAddresses();
+
+				while (a.hasMoreElements()) {
+					InetAddress addr = a.nextElement();
+					String hostAddress = addr.getHostAddress();
+					if ((!hostAddress.startsWith("127.") 
+							&& !hostAddress.contains(":"))
```
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java]: This is to exclude IPv6? Perhaps add a comment about it.

Comment (ChielBruin) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java]: Yes it is. I agree with adding a comment for that.

File: src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+	//Use eager initialization of the singleton.
+	private static final QRGenerator INSTANCE = new QRGenerator();
+
+	//Location and name of QR image.
+	public final String location = "qrcode.png";
+	//Width of QR image.
+	public final int width = 250;
+	//Height of QR image.
+	public final int heigth = 250;
+
```
Comment (Taeir) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java:40]: These fields could be constants (public static final)

File: src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java (hunk 3 above)
Comment (Diocruel) [inline src/main/java/nl/tudelft/contextproject/qrgenerator/QRGenerator.java]: Added a comment to the javadoc of the method

**Comment by Taeir** [comment]
Looks good now you have resolved my comments. Can be merged when travis is done.

**Pull request merged**